src/DynamoCoreWpf/Controls/IncanvasSearchControl.xaml.cs

```diff
@@ -29,28 +27,13 @@ public SearchViewModel ViewModel
             get { return DataContext as SearchViewModel; }
         }
 
-        private WorkspaceView workspaceView;
-        private DynamoView dynamoView;
-
         public InCanvasSearchControl()
         {
             InitializeComponent();
-
-            Loaded += (sender, e) =>
+            if (Application.Current != null)
             {
-                if (workspaceView == null)
-                {
-                    workspaceView = WpfUtilities.FindUpVisualTree<WorkspaceView>(Parent);
-                }
-
-                if (dynamoView != null) return;
-
-                dynamoView = WpfUtilities.FindUpVisualTree<DynamoView>(Parent);
-                if (dynamoView != null)
-                {
-                    dynamoView.Deactivated += (s, args) => { OnRequestShowInCanvasSearch(ShowHideFlags.Hide); };
-                }
-            };
+                Application.Current.Deactivated += (s, args) => { OnRequestShowInCanvasSearch(ShowHideFlags.Hide); };
+            }
```

Comment:
IncanvasSearchControl in context menu popup is specified inside template, so `WpfUtilities.FindUpVisualTree` does not find `WorkspaceView` as a parent. 
Actually, there was the same issue with former context menu, but `ContextMenu` control has built-in mechanism to close when window is deactivated